src/gt4py/next/program_processors/runners/dace_fieldview/transformations/util.py

```diff
@@ -8,153 +8,114 @@
+
+        for state in nsdfg.states():
+            for dnode in state.data_nodes():
+                if dnode.data in not_modify_lifetime:
+                    continue
+
+                if dnode.data in nsdfg.constants_prop:
+                    not_modify_lifetime.add(dnode.data)
+                    continue
 
-    def next_nodes(node: dace_nodes.Node) -> Iterable[dace_nodes.Node]:
-        return (
-            (edge.src for edge in graph.in_edges(node))
-            if reverse
-            else (edge.dst for edge in graph.out_edges(node))
-        )
+                desc = dnode.desc(nsdfg)
+                if not desc.transient or type(desc) not in {dace.data.Array, dace.data.Scalar}:
+                    not_modify_lifetime.add(dnode.data)
+                    continue
+                if desc.storage == dace.StorageType.Register:
+                    not_modify_lifetime.add(dnode.data)
+                    continue
 
-    if reverse:
-        begin, end = end, begin
+                if desc.lifetime == dace.AllocationLifetime.External:
+                    not_modify_lifetime.add(dnode.data)
+                    continue
 
-    to_visit: list[dace_nodes.Node] = [begin]
-    seen: set[dace_nodes.Node] = set()
+                try:
+                    # The symbols describing the total size must be a subset of the
+                    #  free symbols of the SDFG (symbols passed as argument).
+                    # NOTE: This ignores the renaming of symbols through the
+                    #   `symbol_mapping` property of nested SDFGs.
+                    if not set(map(str, desc.total_size.free_symbols)).issubset(fsyms):
+                        not_modify_lifetime.add(dnode.data)
+                        continue
+                except AttributeError:  # total_size is an integer / has no free symbols
+                    pass
 
-    while len(to_visit) > 0:
-        node: dace_nodes.Node = to_visit.pop()
-        if node != end and node not in seen:
-            to_visit.extend(next_nodes(node))
-        seen.add(node)
+                # Make it persistent.
+                modify_lifetime.add(dnode.data)
 
-    # If `end` was not found we have to return `None`  to indicate this.
-    if end not in seen:
-        return None
+        # Now setting the lifetime.
+        result[nsdfg.cfg_id] = modify_lifetime - not_modify_lifetime
+        for aname in result[nsdfg.cfg_id]:
+            nsdfg.arrays[aname].lifetime = dace.AllocationLifetime.Persistent
 
-    # `begin` and `end` are not included in the output set.
-    return seen - {begin, end}
+    return result
 
 
-def find_downstream_consumers(
-    state: dace.SDFGState,
-    begin: dace_nodes.Node | dace_graph.MultiConnectorEdge[dace.Memlet],
-    only_tasklets: bool = False,
-    reverse: bool = False,
-) -> set[tuple[dace_nodes.Node, dace_graph.MultiConnectorEdge[dace.Memlet]]]:
-    """Find all downstream connectors of `begin`.
+def gt_find_constant_arguments(
```

Comment:
It is not used, right? or do you plan to call it from the backend workflow?

Reply:
Currently I use it externally in my benchmarking tool, but currently it is not used inside GT4Py, but this is something that the orchestrator for example will need.

The reason this function exists is, because the name of the size and stride arguments are in fact an implementation details.
Furthermore, also for the composition of the arguments, i.e. extracting the sizes and strides from the argument, you also need help from GT4Py.
So I do not see why this function should not exists.

However, I agree that this is not communicated in a nice way.
How about adding a note to the auto optimizer?

Reply:
I do not disagree that we should have this function, it is a very good funtion. Just I would like to see it be called in the dace backend workflow, probably in the decorator function on the first call (`if not sdfg_program._lastargs`).

But maybe better to do it in a separate PR, after the old backend is removed.